tests/integration_tests/test_deploy_model.py

```diff
@@ -0,0 +1,88 @@
+import unittest
+import subprocess
+import os
+import requests
+import time
+import platform
+import signal
+import tempfile
+import shutil
+
+
+class TestDeployModel(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestDeployModel, self).__init__(*args, **kwargs)
+        self.cwd = os.getcwd()
+        self.tabpy_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+        if platform.system() == 'Windows':
+            self.py = 'python'
+        else:
+            self.py = 'python3'
+
+    def setUp(self):
+        prefix = '_TestDeployModel_'
+        self.state_dir = tempfile.mkdtemp(prefix=prefix, dir=self.cwd)
+        self.state_file = open(os.path.join(self.state_dir, 'state.ini'), 'w+')
```

Comment:
I'd always advise using a context manager instead of explicitly trying to do open...close with a file handle.

Reply:
Since all of the other unit tests use open/close I chose to follow the same pattern. However I think we have plans to revamp our tests in the near future so hopefully we can address it then.